react/react2/week1/src/Profile.js

```diff
@@ -0,0 +1,65 @@
+import React, {useState, useContext}  from 'react'
+import UsernameList from './UsernameList';
+import { UserProvider, UserContext } from './UserContext';
+
+const Profile = () => {
+    const [username, setUsername] = useState("");
+    const [usernames, setUsernames] = useState([]);
+/*     const [usernames, setUsernames] = useState([]);
+ *//*  const [repositories, setRepositories] = useState([]);
+ */ const [isLoading, setIsLoading] = useState(false)
+  
+    /* const onChangeHandler = e => {
+      setUsername(e.target.value);
+    }; */
+  
+    const submitHandler = async e => {
+        setUsername(e.target.value);
+
+        setIsLoading(true) 
+     try {
+      e.preventDefault();
+  
+      const profiles = await fetch(`https://api.github.com/search/users?q=${username}`);
+      const profilesJson = await profiles.json();
```

Comment:
I would like to see you handle any error messages the Github API might return.

You will quickly hit the maximum request limit Github imposes. You can see it happen if you open the Chrome Dev Tools and go to the Network tab while searching with your application here.

When you hit the max request limit Github will respond with a 403 status and an error message. You can check if the response you get from the fetch has an `.ok` property. If not that means that the request was unsuccessful and that an error message is included with the response.

Have a look the example I have added below and see if you can handle and display any errors returned.

```
const profiles = await fetch(`https://api.github.com/search/users?q=${username}`);

**if (!profiles.ok) {
  const error = await profiles.json();
  console.log(error.message);
  // ... continue from here :)
}**

const profilesJson = await profiles.json();
```